src/Npgsql/NpgsqlBinaryExporter.cs

```diff
@@ -320,8 +320,10 @@ PgConverterInfo CreateConverterInfo(Type type, NpgsqlDbType? npgsqlDbType = null
             }
             var info = options.GetTypeInfo(type, pgTypeId)
                        ?? throw new NotSupportedException($"Reading is not supported for type '{type}'{(npgsqlDbType is null ? "" : $" and NpgsqlDbType '{npgsqlDbType}'")}");
+
             // Binary export has no type info so we only do caller-directed interpretation of data.
-            return info.Bind(new Field("?", info.PgTypeId!.Value, -1), DataFormat.Binary);
+            return info.Bind(new Field("?",
+                info.PgTypeId ?? ((PgResolverTypeInfo)info).GetDefaultResolution(null).PgTypeId, -1), DataFormat.Binary);
```

Comment:
We need to do this GetDefaultResolution crap as we do not know what we should pick otherwise. (COPY being untyped)